sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -219,7 +221,13 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
       if (extraOptions.get("queryName").isEmpty) {
         throw new AnalysisException("queryName must be specified for memory sink")
       }
-
+      outputMode match {
+        case Append | Complete => // allowed
+        case Update =>
+          throw new AnalysisException("Update ouptut mode is not supported for memory format")
```

Comment:
also "output".  Thinking a little more, I wonder if it would be better to say what is supported?